feat(timeline): Add support for color and colormaps to broken bar plots .

mikelgg93 opened the pull request:
# PR - Add color and colormaps to broken bars.

## Motivation
Many white bars can be difficult to interpret at a glance, without looking at the labels, specially since we do not have hierarchy yet. By adding color to the broken bars and the possibility to add colormap based values, it makes it more discoverable/ easy to interpret.

**Use cases:** 
- Show number of detections at a glance or whether an object is being gazed.
- Mimic enrichments bar on Cloud as surface detected or gazed.

## Proposed Solution

**Example:**
<img width="1712" height="1140" alt="Screenshot 2025-12-16 at 12 57 43" src="https://github.com/user-attachments/assets/00c8edf3-3096-4530-b6a6-7143fe3aa8d9" />

## Key Changes
- Accept data with optional Z-values for color mapping.
- Implemented `_resolve_bar_colors` to determine pen and brush colors based on values.

domstoppable commented on pupil_src/shared_modules/audio_playback.py:
```diff
@@ -178,7 +178,12 @@ def _setup_output_audio(self):
+            import traceback
 
+            logger.warning("Audio device could not be started")
+            logger.debug(traceback.format_exc())
```
This is a very reasonable addition, and if it fixes your problem, I'm happy to merge :)

Can I just ask for one extra line feed to separate the next function?

dip-c replied:
Thanks for the feedback, here you go!

mikelgg93 committed:
feat(timeline): Add support for color and colormaps to broken bar plots .

- Accept data with optional Z-values for color mapping.
- Implemented `_resolve_bar_colors` to determine pen and brush colors based on values.
